services/src/main/scala/cromwell/pipeline/service/ProjectSearchService.scala

```diff
@@ -0,0 +1,141 @@
+package cromwell.pipeline.service
+
+import cromwell.pipeline.datastorage.dto._
+import cromwell.pipeline.model.wrapper.UserId
+import cromwell.pipeline.service.ProjectConfigurationService.Exceptions.NotFound
+import cromwell.pipeline.service.ProjectSearchService.Exceptions._
+import cromwell.pipeline.service.exceptions.ServiceException
+
+import scala.concurrent.{ ExecutionContext, Future }
+import scala.util.{ Failure, Success }
+
+trait ProjectSearchService {
+
+  def searchProjects(request: ProjectSearchRequest, userId: UserId): Future[ProjectSearchResponse]
+
+}
+
+object ProjectSearchService {
+
+  object Exceptions {
+    sealed abstract class ProjectSearchException(message: String) extends ServiceException(message)
+    final case class InternalError(message: String = "Internal error") extends ProjectSearchException(message)
+  }
+
+  // scalastyle:off method.length
+  def apply(
+    projectService: ProjectService,
+    projectFileService: ProjectFileService,
+    projectConfigurationService: ProjectConfigurationService
+  )(
+    implicit ec: ExecutionContext
+  ): ProjectSearchService =
+    new ProjectSearchService {
+
+      override def searchProjects(request: ProjectSearchRequest, userId: UserId): Future[ProjectSearchResponse] = {
+        val eventualUserProjects = projectService.getUserProjects(userId).recoverWith {
+          case _ => internalError("fetch projects")
+        }
```

Comment:
[minor] do you still need it here? It only causes confusion as you've already seen. Can't you just inline this variable at the only usage?

Reply:
Removed